### ISSUE TITLE
`const` variables declared not in global scope don't seem to be respecting the `const` keyword

### ISSUE THREAD

**Issue by mohammadfawaz** [FuelLabs/sway#2076]
The following compiles but `X` is not actually constant evaluated. If we move the definition of `X` to the global scope, then we get an error saying `Could not evaluate initializer to a const declaration.` as expected.
```rust
script;

fn bla(x: u64) -> u64 {
    x + 1
}

fn main() -> u64 {
    const X = bla(0);
    X
}
```
It is possible that this behavior is desirable. If so, we should properly document it. If not, then this is a bug.

**Comment by otrho** [comment]
I don't think we want this -- just use a `let` if you need a local non-const var.  This might be related to the way we use `TypedVariableDeclaration` instead of `TypedConstDeclaration` ala #1957.
